src/Microsoft.AspNetCore.Razor.Language/RazorProjectEngine.cs

```diff
@@ -47,10 +59,26 @@ public virtual RazorCodeDocument ProcessDesignTime(RazorProjectItem projectItem)
             return codeDocument;
         }
 
+        internal virtual RazorCodeDocument ProcessDesignTime(RazorProjectItem projectItem, IReadOnlyList<TagHelperDescriptor> tagHelpers)
```

Comment:
I'm not a super fan of this because it only solves the problem with tag helpers. We need to solve it for *ALL** of the source files as well as the tag helpers.

What if we just make `CreateCodeDocumentDesignTime` `protected internal` ?

Reply:
> What if we just make CreateCodeDocumentDesignTime protected internal ?

I considered it but `protected internal` is protected OR internal which means this would be public. Do we want to expose this?

Reply:
> ALL* of the source files

Could you expand on that? Do you mean imports?

Reply:
No I still think we are on the same page. That is what I envisioned for imports - It directly taking in a list of `RazorSourceDocument`. As I commented on the issue for this, I was going to do that separately. As part of adding imports tracking to `DocumentSnapshot`. But if you think it is worth having ProjectEngine support for imports in this PR, I'll update it. Let me know if I still don't make any sense.

Reply:
Sure I'll update it. But for now the only change needed in `DocumentGeneratedOutputTracker` to make tag helpers work was https://github.com/aspnet/Razor/pull/2346/files#diff-277383bfb275d15d91a911ee7b04ee3bR95.